src/Microsoft.Diagnostics.Monitoring.StartupHook/Exceptions/Pipeline/Steps/ExceptionEventsPipelineStep.cs

```diff
@@ -112,6 +112,24 @@ private static ReadOnlySpan<StackFrame> ComputeEffectiveCallStack(Exception exce
                 index++;
             }
 
+            if (index >= threadStackTrace.FrameCount)
```

Comment:
Instead of iterating the stack a second item, update the original code to remove the IL offset check so it finds the first matching frame solely based on the method, and then unconditionally swap out the frame.